src/debt_order/index.ts

```diff
@@ -73,7 +58,14 @@ export class DebtOrder {
         return this.dharma.order.checkOrderFilledAsync(this.debtOrderData);
     }
 
-    public async fill(): Promise<string> {
+    public async fill(creditorAddress: string): Promise<string> {
```

Comment:
I would keep the decomposed functions and just mark them as `private`

Reply:
Doesn't there need to be an interface for this? Like `FillParameters`?